### PR TITLE
version now is release date and build is build number

### PR DESCRIPTION
**Pull request by nadeemyaseen-rs**
Make version as the release date of the product and build as semantic version number. Now the output of --version will look like:

image

### CODE REVIEW

**Comment by nadeemyaseen-rs** [comment]
@ravic-rs  and @volodymyrkochyn  kindly go through the changes and let me know if they affect anything else. If not then I will take it to Raptor level.

**Comment by KochynVolodymyr** [comment]
What is the ticket number?
FOEDAG version affects About widget and project migration process.

File: foedag_version.h.in [hunk 1]
```diff
@@ -3,4 +3,5 @@
 #define FOEDAG_VERSION      @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@
```
Comment (KochynVolodymyr) [inline foedag_version.h.in:3]: It is make sense to rename this to FOEDAG_BUILD since it suppose to represent build but not version

**Comment by ravic-rs** [comment]
> > What is the ticket number?
> 
> 
> 
> EDA-1883
> 
> 

No hyperlinks to internal pages please.

**Comment by ravic-rs** [comment]
Foedag
Version : 2023.08
Build     : 0.0.231
Hash     : bd98a7ab
Date      : Aug 23 2023
Type      : Engineering

**Comment by alain-rs** [comment]
@nadeemyaseen-rs @ravic-rs , let's finish this PR.

File: foedag_version.h.in (hunk 1 above)
Comment (ravic-rs) [reply foedag_version.h.in]: Fixed this.

**Comment by ravic-rs** [comment]
@KochynVolodymyr we will have to update the "About" display later.

**Pull request merged**